### PR TITLE
Step painter

### PR DESCRIPTION
**Pull request by theis999**
Demo showing the effect of painting rows in the step list.

A colour picker was added to the auto put panel, mainly due to the ample amount of space there.
An event handler for updating the colour value of the colour picker is implemented.
An event handler for updating the background colour of selected rows in the step list.

To complete, the colour of rows needs to be stored in the save file.
Due note that this is built on #180

### CODE REVIEW

**Comment by theis999** [comment]
image

**Comment by MortenTobiasNielsen** [comment]
I guess it cannot hurt to add this, so the user has the possibility to use it. :)

**Comment by theis999** [comment]
I have added **save/load** functionality

In the switch case of step parameters i also added [[[likely]]](https://en.cppreference.com/w/cpp/language/attributes/likely) which is something i stumbled upon that should optimize the compilation for those cases.

File: src/OpenTas.cpp [hunk 1]
```diff
@@ -163,6 +165,7 @@ Category OpenTas::extract_steps(std::ifstream& file, DialogProgressBar* dialog_p
 		step.Orientation = Capitalize(segments[5]);
 		step.Direction = Capitalize(segments[6]);
 		step.Comment = comment;
+		step.Colour = segments[10];
```
Comment (MortenTobiasNielsen) [inline src/OpenTas.cpp:168]: Colour won't be in the current save files, so this will crash the application.

**Review by MortenTobiasNielsen — CHANGES_REQUESTED** [review]
You will have to handle colour in the same way as I have handled comment, so it is only extracted if it is there.

**Comment by theis999** [comment]
Alright, i changed some stuff and tested on file without colours, and one without colours and comments.

**Review by MortenTobiasNielsen — APPROVED** [review]

**Pull request merged**